### PR TITLE
r/instance: Function for availability zone config

### PR DESCRIPTION
**Pull request by YakDriver**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates #12995 

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
NONE
```

Output from GovCloud acceptance testing:

```
--- PASS: TestAccAWSInstanceDataSource_privateIP (97.68s)
--- PASS: TestAccAWSInstanceDataSource_VPC (117.44s)
--- PASS: TestAccAWSInstanceDataSource_VPCSecurityGroups (130.15s)
```

Output from commercial acceptance testing:

```
--- PASS: TestAccAWSInstanceDataSource_privateIP (129.91s)
--- PASS: TestAccAWSInstanceDataSource_VPC (124.76s)
--- PASS: TestAccAWSInstanceDataSource_VPCSecurityGroups (142.31s)
```

### CODE REVIEW

File: aws/resource_aws_instance_test.go [hunk 1]
```diff
@@ -3280,6 +3280,19 @@ data "aws_availability_zones" "available" {
 `, strings.Join(excludeZoneIds, "\", \""))
 }
 
+func testAccAvailableAZsNoOptInConfig() string {
```
Comment (anGie44) [inline aws/resource_aws_instance_test.go:3283]: nice! we could also document this function in `https://github.com/terraform-providers/terraform-provider-aws/blob/master/docs/contributing/running-and-writing-acceptance-tests.md` for contributors to re-use

File: aws/resource_aws_instance_test.go [hunk 2]
```diff
@@ -4886,16 +4899,7 @@ resource "aws_subnet" "test" {
...
-data "aws_availability_zones" "available" {
-  state            = "available"
-
-  filter {
-    name   = "opt-in-status"
-    values = ["opt-in-not-required"]
-  }
-}
-
+	return testAccAvailableAZsNoOptInConfig() + fmt.Sprintf(`
```
Comment (anGie44) [inline aws/resource_aws_instance_test.go:4902]: Nit: could also use `composeConfig` to concatenate these 2 configs

File: aws/resource_aws_instance_test.go (hunk 1 above)
Comment (YakDriver) [reply aws/resource_aws_instance_test.go:3283]: I added docs on this!

File: aws/resource_aws_instance_test.go (hunk 2 above)
Comment (YakDriver) [reply aws/resource_aws_instance_test.go]: Switched to use this!

**Pull request merged**